### DISCUSSION TITLE
How to switch rendering engine to Direct2D?

### DISCUSSION THREAD

**Discussion by Naylor55** [AvaloniaUI/Avalonia#14209] [Q&A] [answered]
env
* Avalonia.Desktop : 11.0.6
* Avalonia : 11.0.6
* Avalonia.Diagnostics : 11.0.6
* Avalonia.Fonts.Inter : 11.0.6
* Avalonia.Themes.Fluent : 11.0.6
* Avalonia.Direct2D1  11.0.6
* CommunityToolkit.Mvvm ：8.2.2

In order to switch the rendering engine to Direct2D, I found the following solution on Google, but they all failed。

*  first , add Avalonia.Direct2D1 to  project
*  second，config the  app.axaml.cs

~~~

public override void Initialize()
    {
        AvaloniaLocator.CurrentMutable
            .Bind<IRenderLoop>().ToConstant(new Direct2DRenderLoop())
            .Bind<IPlatformRenderInterface>().ToConstant(new DefaultRenderInterface())
            .Bind<IRenderTimer>().ToConstant(new DefaultRenderTimer());
        base.Initialize();
    }

~~~

* after that，IDE(VS2022)  ，It indicates that  did not find CurrentMutable

**Comment by timunie** [comment, marked as answer]
try: 

```diff
    public static AppBuilder BuildAvaloniaApp()
        => AppBuilder.Configure<App>()
            .UsePlatformDetect()
+           .UseDirect2D1()
            .WithInterFont()
            .LogToTrace()
            .UseReactiveUI();
```

**Comment by Naylor55** [comment]
@timunie year! , it work!
